### DISCUSSION TITLE
Release notes

### DISCUSSION THREAD

**Discussion by asarkar** [open-telemetry/opentelemetry-java#2887] [Q&A]
The release notes about [Version 0.17.0 (RC)](https://github.com/open-telemetry/opentelemetry-java/releases) don't say what the actual version is (I'm guessing `0.17.0`), or where those libraries are available. As of this writing, they aren't in Maven Central. The [link for discussions](https://github.com/opentelemetry/opentelemetry-java/discussions) is broken, which is mind boggling, since it's part of the release announcement. Maven Central has a version `0.16.1`, which has no corresponding release notes.

Simply knowing how to get and use a version doesn't have to be a painstaking project. Can we expect the release notes to have been reviewed by someone other than the author?

**Comment by jkwatson** [comment]
Hi @asarkar . I'm sorry you're having trouble getting the dependencies via maven central. I assure you they are there currently. And, our main README has been updated to point to the latest versions, with sample dependencies for both maven and gradle.

**Reply by jkwatson** [reply]
I've updated the release notes to have the non-broken links to the discussions. Thanks for pointing it out!

**Comment by jkwatson** [comment]
I'm not sure what you're looking at with the version `0.16.1`. This project has never published that version. Are you sure you're not looking for the instrumentation project, which hasn't published its `0.17.0` version yet?

**Reply by asarkar** [reply]
I'm using the following dependencies and am confused what versions are compatible.
```
io.opentelemetry.instrumentation:opentelemetry-instrumentation-api
io.opentelemetry.instrumentation:opentelemetry-grpc-1.5
io.opentelemetry.instrumentation:opentelemetry-logback-1.0.0

io.opentelemetry:opentelemetry-sdk-tracing
io.opentelemetry:opentelemetry-extension-kotlin
```

I think a compatibility table between the two groups would be _very_ useful, especially given all the breaking changes.

**Reply by jkwatson** [reply]
At the moment, due to the breaking changes, the instrumentation libraries must use the *same* major/minor version of the api/sdk libraries that they use.

**Reply by jkwatson** [reply]
A compatibility table would definitely be an issue/discussion for the instrumentation repository, though, not here.  :)

**Reply by asarkar** [reply]
Ok, so `0.16.0` would be the common latest version? Do you see why this can be a complicated endeavor for a user?

**Reply by jkwatson** [reply]
Looks like keeping them in sync is documented here over in that project:  https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/manual-instrumentation.md#dependencies

I'm sorry this is difficult. We're still pre-release on this project. If you would like to help us get the documentation better for our upcoming 1.0 release, that would be greatly appreciated.

**Reply by asarkar** [reply]
> help us get the documentation better

I'm happy to help.

> prior to version 1.0.0, opentelemetry-javaagent-all.jar only supports manual instrumentation using the opentelemetry-api version with the same version number as the Java agent you are using.

I'm not using the Java agent jar. I see that someone can come to the conclusion that the two groups must use the same version for libraries, but it's a stretch for most people. Not to mention finding such a disclaimer in a page for manual instrumentation is finding a needle in a haystack.
I'll open a discussion on that project. Thank you for the clarification provided.

**Comment by asarkar** [comment]
Opened https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2337 for compatibility table.

**Comment by asarkar** [comment]
@jkwatson [io.opentelemetry:opentelemetry-sdk-tracing](https://search.maven.org/artifact/io.opentelemetry/opentelemetry-sdk-tracing) has no `0.16.0` version. I appreciate what the community is doing on this project, but the versioning looks like a complete mess.

**Reply by jkwatson** [reply]
That's because we renamed that artifact many versions ago.
https://search.maven.org/search?q=a:opentelemetry-sdk-trace

**Reply by jkwatson** [reply]
If you use the `opentelemetry-bom`, all your versions should be kept in sync, and everything should be available.

**Reply by jkwatson** [reply]
Here is the easiest way to see all the current non-alpha libraries:  https://search.maven.org/artifact/io.opentelemetry/opentelemetry-bom/0.17.0/pom

**Reply by jkwatson** [reply]
And if you want the alpha modules: 
https://search.maven.org/artifact/io.opentelemetry/opentelemetry-bom-alpha/0.17.0-alpha/pom
